### PR TITLE
Added docs for setting up TLS certificates

### PR DESCRIPTION
**Pull request by owais-rehman**

### CODE REVIEW

**Review by OsamaKhan220 — DISMISSED** [review]

File: content/for-administrators/how-to-guides/certificate-management/tls-certs.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+This document will explain a step-by-step approach of configuring TLS certificates for different tenants using Infra GitOps.
+
+1. First step is to navigate to correct path inside your Infra GitOps repository. For this example we are using following path:
+`<environment>/tenant-operator-config/templates/`
+Here `<environment>` correspond to the cluster where you want to deploy this.
+1. In this directory we need to create 2 resources [`Template`](https://docs.stakater.com/mto/main/crds-api-reference/template.html) and [`TemplateGroupInstance`](https://docs.stakater.com/mto/main/crds-api-reference/template-group-instance.html). A detailed explanation about why each resource is needed and what are the contents of each resource is given below:
+
+- ## Template
+
+    This resource is reponsible for keeping a record (template) or underlying resources (YAML files) that needs to be deployed to tenant namespaces.
```
Comment (karl-johan-grahn) [inline content/for-administrators/how-to-guides/certificate-management/tls-certs.md:12]: just wonder why you indent this text?

Comment (owais-rehman) [reply content/for-administrators/how-to-guides/certificate-management/tls-certs.md:12]: The reason for indenting this text was to cover this text in current bullet point under headings

File: content/for-administrators/how-to-guides/certificate-management/tls-certs.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+# Configuring TLS Certificates Using Infra GitOps
+
+This document provides a step-by-step guide to configure TLS certificates for different tenants using Infra GitOps.
+
+## Step 1: Navigate to the Target Path
+
+Navigate to the appropriate path in your Infra GitOps repository. For this example, the path is:
+
+```plaintext
+<environment>/tenant-operator-config/templates/
```
Comment (rasheedamir) [inline content/for-administrators/how-to-guides/certificate-management/tls-certs.md:10]: @owais-rehman this should be called cluster then and not environment; as environment is dev, stage and prod etc.

File: content/for-developers/how-to-guides/expose-applications-to-internet/expose-applications-to-internet.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    host: <desired-dns-name>
+    path: <desired-path>
+```
+
+#### Important Details
+
+- **Annotations**:
+    - `cert-utils-operator.redhat-cop.io/certs-from-secret`: Specifies the name of the secret that stores the TLS certificate created by the Certificate resource.
+    - `external-dns.alpha.kubernetes.io/hostname`: Registers the DNS record with the configured provider (e.g., Cloudflare).
+    - `cert-utils-operator.redhat-cop.io/inject-CA`: Indicates whether to inject the Certificate Authority (CA) into the Route. Set to "false" if not required.
```
Comment (rasheedamir) [inline content/for-developers/how-to-guides/expose-applications-to-internet/expose-applications-to-internet.md:38]: when will this be required @owais-rehman ?

File: content/for-administrators/how-to-guides/certificate-management/tls-certs.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
+# Configuring TLS Certificates and External DNS
+
+This document provides a step-by-step guide to configure TLS certificates and External DNS for different tenants.
+
+## Step 1: Setup DNS creds in Vault
+
+Go to `common-shared-secret` path in Vault and create a secret `external-dns-creds`. This secret mainly have credentials for authenticating with DNS provider (in present case `Cloudflare`) and should contain following fields:
```
Comment (rasheedamir) [inline content/for-administrators/how-to-guides/certificate-management/tls-certs.md:7]: we can improve this

**Review by rasheedamir — APPROVED** [review]

**Pull request merged**